paddle/pybind/pybind.cc

```diff
@@ -25,9 +26,143 @@ namespace pd = paddle::framework;
 
 USE_OP(add_two);
 
+struct PlaceDebugString : public boost::static_visitor<std::string> {
+  std::string operator()(const paddle::platform::GPUPlace& place) const {
+    return "GPU(" + std::to_string(place.device) + ")";
+  }
+
+  std::string operator()(const paddle::platform::CPUPlace& place) const {
+    return "CPU";
+  }
+};
+
+template <typename T>
```

Comment:
The `static if` is added in [C++ 17](https://medium.com/@LoopPerfect/c-17-vs-c-14-if-constexpr-b518982bb1e2)